raster-foundry/raster-foundry
# Build and ship HITL container in CI; Kickoff HITL job in POST

## Overview

This PR:
- makes the HITL prediction work a CLI and have it installed in the `batch-hitl` container so that it can run at the container entry point
- supports specifying a S3 URL for prediction output and other artifacts
- adds support for building and shipping the HITL container in CI/CD process
- modifies the POST to `/api/hitl-jobs` so that the HITL batch job is kicked off in AWS Batch when in staging/prod

### Checklist

- ~New tables and queries have appropriate indices added~
- ~Any new SQL strings have tests~
- ~Any new endpoints have scope validation and are included in the integration test csv~

## Testing Instructions

- Create and push up a `test/...` branch to trigger a staging deploy because we want to make sure that the prediction can run in a GPU enabled batch compute environment in AWS
- Look for Slack messages, and after a staging deploy is done, if you have access to the deployment repo and can apply infra change to staging, run instructions in here: https://github.com/azavea/raster-foundry-deployment/pull/296; if not, ping me and I can pair with you.

Closes https://github.com/azavea/foss4g-2022-hitl-groundwork-rastervision/issues/17
